### DISCUSSION TITLE
How to obtain a field value from devtools command with a script

### DISCUSSION THREAD

**Discussion by aitzkora** [qutebrowser/qutebrowser#8012] [Q&A] [answered]
Hi,
to connect to a instant messenger service I need to get a token for which I can get the value launching `:devtools` and make a copy paste. Is it possible to script that process, using js or a qutebrowser command ? 
Thanks, by advance

**Comment by toofar** [comment, marked as answer]
I don't think there is a good way. How much do you want to automate? If you don't want to be using the browser at all probably go with a proper browser automation framework. If you are going to be using it in qute anyway, uh. If the token is stored in a cookie or local or session storage there might be some way to exfiltrate it. Eg use :jseval to grab a value and send it to something listening on localhost. Or if you are on Qt 6.5+ you may be able to [look at request headers](https://doc.qt.io/qt-6/qwebengineurlrequestinfo.html#httpHeaders) in the request interceptor.

**Comment by aitzkora** [comment]
Hi @toofar , thanks for your reply. Honestly, a first solution as you mention with jseval seems to be correct. I dont like to use my mouse in devtools. If a could grab a value and put it in the clipboard (or a file) it would be nice. I will adapt the following code 
```javascript

function getCookieValue(key) {
    var cookies = document.cookie.split('; ');
    alert(cookies);
    for (var i = 0, parts; (parts = cookies[i] && cookies[i].split('=')); i++) {
          if (decode(parts.shift()) === key) {
                  return decode(parts.join('='));
                }
        }
    return null;
}

function decode(s) {
    return decodeURIComponent(s.replace(/\+/g, ' '));
}

alert(getCookieValue("MYCOOKIE"));
```
taken there https://stackoverflow.com/questions/12059958/get-a-cookies-single-value-javascript
